internal/zaputil/logrus.go

```diff
@@ -0,0 +1,96 @@
+// Copyright 2021, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package zaputil
+
+import (
+	"fmt"
+	"github.com/sirupsen/logrus"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+	"io/ioutil"
+	"sync"
+)
+
+type zapWrapper zap.Logger
+
+type redirect struct {
+	from *logrus.Logger
+	to *zap.Logger
+}
+
+var (
+	mu        sync.Mutex
+	redirects = make(map[redirect]bool)
+	levelsMap = newLevelsMap()
+)
+
+// RedirectLogrusLogs mutes the output of the supplied logrus.Logger and adds a hook to it.
+// When fired the hook creates a zap entry using the supplied logrus entry.
+func RedirectLogrusLogs(from *logrus.Logger, to *zap.Logger) {
+	mu.Lock()
+	defer mu.Unlock()
+	r := redirect{from, to}
+	if redirects[r] {
```

Comment:
nit: since the boolean isn't really used and only an existence check is performed I believe it's customary to do something like:
```go
_, ok := redirects[r] ; if ok {
  return
}
```

Reply:
Yes, you are correct. I was thinking since the map returns true only if the key exists and the value is true, let me be terse. Plus, think it's even better to use an empty struct instead of bool.

Reply:
I think it's ok. `Using a map[string]bool` is pretty common when you're using it as a set so you can do `if redirects[r] { ... }`.